lib/utilities/pipeline.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+var Promise = require('rsvp').Promise;
+var _       = require('lodash-node');
+
+var chalk = require('chalk');
+var blue  = chalk.blue;
+var red   = chalk.red;
+var grey  = chalk.grey;
+
+function Pipeline(hookNames, options) {
```

Comment:
I think using core object would make it easier for other ember-devs to join in. It's not hard with a prototype but when being used to the ember.object model I think it's easier to other devs when being able to work with a 'normal' object.